src/CattyTests/DataModel/VariablesContainerTest.swift

```diff
@@ -365,4 +365,124 @@ final class VariablesContainerTest: XCTestCase {
+        XCTAssertTrue(result)
+
+        result = container.addObjectVariable(userVariable2, for: objectB)
+        XCTAssertTrue(result)
+
+        container.programVariableList.add(userVariable3 as Any)
+
+        XCTAssertTrue(container.getUserVariableNamed(userVariable1?.name, for: objectA)?.isEqual(to: userVariable1) == true)
+        XCTAssertTrue(container.getUserVariableNamed(userVariable1?.name, for: objectB) == nil)
+        XCTAssertTrue(container.getUserVariableNamed(userVariable3?.name, for: objectA)?.isEqual(to: userVariable3) == true)
+        XCTAssertTrue(container.getUserVariableNamed(userVariable3?.name, for: objectB)?.isEqual(to: userVariable3) == true)
+        XCTAssertTrue(container.getUserVariableNamed(userVariable4?.name, for: objectA) == nil)
+    }
+
+    func testGetUserListNamed() {
+        let objectA = SpriteObject()
+        objectA.name = "testObjectA"
+
+        let objectB = SpriteObject()
+        objectB.name = "testObjectB"
+
+        let list1 = UserVariable(name: "testName1", isList: true)
+        let list2 = UserVariable(name: "testName2", isList: true)
+        let list3 = UserVariable(name: "testName3", isList: true)
+        let list4 = UserVariable(name: "testName4", isList: true)
+
+        let container = VariablesContainer()
+
+        var result = container.addObjectList(list1, for: objectA)
+        XCTAssertTrue(result)
+
+        result = container.addObjectList(list2, for: objectB)
+        XCTAssertTrue(result)
+
+        container.programListOfLists.add(list3 as Any)
+
+        XCTAssertTrue(container.getUserListNamed(list1?.name, for: objectA)?.isEqual(to: list1) == true)
+        XCTAssertTrue(container.getUserListNamed(list1?.name, for: objectB) == nil)
+        XCTAssertTrue(container.getUserListNamed(list3?.name, for: objectA)?.isEqual(to: list3) == true)
+        XCTAssertTrue(container.getUserListNamed(list3?.name, for: objectB)?.isEqual(to: list3) == true)
+        XCTAssertTrue(container.getUserListNamed(list4?.name, for: objectA) == nil)
+    }
+
+    func testRemoveUserVariableNamed() {
+        let objectA = SpriteObject()
+        objectA.name = "testObjectA"
+
+        let objectB = SpriteObject()
+        objectB.name = "testObjectB"
+
+        let objectC = SpriteObject()
+        objectC.name = "testObjectC"
+
+        let userVariable1 = UserVariable(name: "testName1", isList: false)
+        let userVariable2 = UserVariable(name: "testName2", isList: false)
+        let userVariable3 = UserVariable(name: "testName3", isList: false)
+
+        let container = VariablesContainer()
+
+        let result = container.addObjectVariable(userVariable1, for: objectA)
+        XCTAssertTrue(result)
+
+        container.programVariableList.add(userVariable2 as Any)
+
+        XCTAssertFalse(container.removeUserVariableNamed(userVariable1?.name, for: objectB))
+        XCTAssertTrue(container.removeUserVariableNamed(userVariable1?.name, for: objectA))
```

Comment:
Could you please also verify that the number of variables in the container decreased by one and that the proper variable has been removed?